app/controllers/schools/public_controller.rb

```diff
@@ -4,13 +4,13 @@ class PublicController < ApplicationController
 
     def create
       authorize! :change_public, @school
-      @school.update!(public: true)
+      @school.update!(public: true, data_sharing: :public)
```

Comment:
For this PR I've keep the admin buttons for toggling public/private. These will be removed in the next PR. I'm still setting the `public` flag to leave it consistent with previous code until its removed.